src/twin/factory_reset.rs

```diff
@@ -99,51 +151,28 @@ impl FactoryReset {
 
         self.ensure()?;
 
-        let restore_paths = match in_json["restore_settings"].as_array() {
-            Some(settings) => {
-                let mut paths = vec![];
-                let mut settings: Vec<&str> =
-                    settings.iter().map(|v| v.as_str().unwrap()).collect();
-
-                // enforce a value exists only once
-                settings.sort();
-                settings.dedup();
-
-                while let Some(s) = settings.pop() {
-                    if self.settings.contains_key(s) {
-                        let path = self.settings.get(s).unwrap();
-
-                        if let Ok(true) = std::path::Path::new(&path).try_exists() {
-                            paths.push(path.clone());
-                        } else {
-                            anyhow::bail!("invalid restore path received: {path}");
-                        }
-                    } else {
-                        anyhow::bail!("unknown restore setting received: {s}");
-                    }
+        if in_json["mode"].as_u64().is_none() {
+            anyhow::bail!("reset mode missing or not supported");
+        }
+        let preserve = in_json["preserve"].as_array();
+        if  preserve.is_some() {
+            let keys=FactoryReset::factory_reset_keys()?;
+            for topic in preserve.unwrap().iter() {
+                let topic = String::from(topic.to_string().trim_matches('"'));
+                if ! keys.contains(&topic) {
+                    anyhow::bail!("unknown preserve topic received: {topic}");
                 }
-
-                paths.join(";")
             }
-            _ => String::from(""),
-        };
-
-        match &in_json["type"].as_u64() {
-            Some(reset_type) => {
-                bootloader_env::set("factory-reset-restore-list", restore_paths.as_str())?;
-                bootloader_env::set("factory-reset", &reset_type.to_string())?;
-
-                self.report_factory_reset_status("in_progress").await?;
-
-                systemd::reboot().await?;
-
-                Ok(None)
-            }
-            _ => anyhow::bail!("reset type missing or not supported"),
         }
+
+        bootloader_env::set("factory-reset", &in_json.to_string())?;
+        self.report_factory_reset_status("in_progress").await?;
+        systemd::reboot().await?;
+        Ok(None)
     }
 
     async fn report_factory_reset_status(&self, status: &str) -> Result<()> {
+        // ToDo why is that not the same format as for the cloud?
```

Comment:
@JanZachmann

Reply:
No special reason and we should make it equal. Pls create a task for omnect-ui to adapt to the new structure.

Reply:
task imho already exists, created by you.